sdk/messaging/azservicebus/receiver.go

```diff
@@ -608,32 +608,35 @@ func (r *Receiver) newReleaserFunc(receiver amqpwrap.AMQPReceiver) func() {
 	}
 
 	ctx, cancel := context.WithCancel(context.Background())
-	done := make(chan struct{})
+	releaseLoopDone := make(chan struct{})
 	released := 0
 
 	// this func gets called when a new ReceiveMessages() starts
 	r.cancelReleaser.Store(func() string {
 		cancel()
-		<-done
+		<-releaseLoopDone
 		return receiver.LinkName()
 	})
 
 	return func() {
-		defer close(done)
+		defer close(releaseLoopDone)
 
 		for {
-			// we might not have all the messages we need here.
 			msg, err := receiver.Receive(ctx, nil)
 
 			if err == nil {
-				err = receiver.ReleaseMessage(ctx, msg)
-			}
+				// We don't use `ctx` here to avoid cancelling Release(), and leaving this message
+				// in limbo until it expires.
+				err = receiver.ReleaseMessage(context.Background(), msg)
```

Comment:
The background context makes me nervous. Should this be its own context with a timeout in case the operation hangs?

Reply:
Agreed. Maybe a 1 minute timeout?

Reply:
Added in a timeout.